packages/x-data-grid/src/hooks/features/rows/useGridParamsApi.ts

```diff
@@ -108,6 +108,9 @@ export function useGridParamsApi(apiRef: React.MutableRefObject<GridPrivateApiCo
 
   const getRowValue = React.useCallback<GridParamsApi['getRowValue']>(
     (row, colDef) => {
+      if (!row) {
+        return null;
+      }
```

Comment:
Why is the defensive check needed here? The typing says `row` should always be defined.